Civi/Token/TokenCompatSubscriber.php

```diff
@@ -59,6 +59,11 @@ public function onRender(TokenRenderEvent $e): void {
       // For historical consistency, we filter out unrecognized tokens.
       return '';
     });
+    // This removes the pattern used in greetings of having bits of text that
+    // depend on the tokens around them - ie '{first_name}{ }{last_name}
+    // has an extra construct '{ }' which will resolve as a space if the
+    // tokens on either side are resolved to 'something'
+    $e->string = preg_replace('/\\\\|\{(\s*)?\}/', ' ', $e->string);
```

Comment:
this was previously being called before we stopped the token processor from calling `replaceGreetingTokens` and `replaceContactTokens`  so it is now being called again. With this merged it will have changed and changed back within the same release